Sources/SMTP.swift

```diff
@@ -220,4 +183,41 @@ public struct SMTP {
             completion?([], mails.map { ($0, error) })
         }
     }
+
+    /// Returns a `Mail`.
+    ///
+    /// - Parameters:
+    ///     - from: The `User` that the `Mail` will be sent from.
+    ///     - to: Array of `User`s to send the `Mail` to.
+    ///     - cc: Array of `User`s to cc. Defaults to none.
+    ///     - bcc: Array of `User`s to bcc. Defaults to none.
+    ///     - subject: Subject of the `Mail`. Defaults to none.
+    ///     - text: Text of the `Mail`. Defaults to none.
+    ///     - attachments: Array of `Attachment`s for the `Mail`. If the `Mail`
+    ///                    has multiple `Attachment`s that are alternatives to
+    ///                    to plain text, the last one will be used as the
+    ///                    alternative (all the `Attachments` will still be
+    ///                    sent). Defaults to none.
+    ///     - additionalHeaders: Additional headers for the `Mail`. Header keys
+    ///                          are capitalized and duplicate keys will
+    ///                          overwrite each other. Defaults to none. The
+    ///                          following will be ignored: CONTENT-TYPE,
+    ///                          CONTENT-DISPOSITION, CONTENT-TRANSFER-ENCODING.
+    public func makeMail(from: User,
+                         to: [User],
+                         cc: [User] = [],
+                         bcc: [User] = [],
+                         subject: String = "",
+                         text: String = "",
+                         attachments: [Attachment] = [],
+                         additionalHeaders: [String: String] = [:]) -> Mail {
+        return Mail(from: from,
+                    to: to,
+                    cc: cc,
+                    bcc: bcc,
+                    subject: subject,
+                    text: text,
+                    attachments: attachments,
+                    additionalHeaders: additionalHeaders)
+    }
```

Comment:
With your changes to the `Mail` struct, I think this extra `makeMail()` function can be removed.